Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
Yes, it would.
As explained in the PR description, there are a few reasons why returning `null` seems to be the most sensible thing to do.

> To sum things up, I believe we're better off simply returning default(T) if we get an OmitSpecimen.

It seems that, using NSubstitute, it is possible for the callback to conditionally setup the member. In Moq, this is not viable.